Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -419,891 +444,969 @@ void CGContextSetCTM(CGContextRef context, CGAffineTransform transform) {
-    CGContextSetRGBFillColor(context, gray, gray, gray, alpha);
+bool CGContextIsPathEmpty(CGContextRef context) {
+    NOISY_RETURN_IF_NULL(context, StubReturn());
+    UNIMPLEMENTED();
 }
 
 /**
  @Status Interoperable
 */
-void CGContextSetRGBFillColor(CGContextRef context, CGFloat r, CGFloat g, CGFloat b, CGFloat a) {
-    NOISY_RETURN_IF_NULL(context);
-    ComPtr<ID2D1SolidColorBrush> brush;
-    FAIL_FAST_IF_FAILED(context->RenderTarget()->CreateSolidColorBrush({ r, g, b, a }, &brush));
-    FAIL_FAST_IF_FAILED(brush.As(&context->CurrentGState().fillBrush));
+CGRect CGContextGetPathBoundingBox(CGContextRef context) {
+    NOISY_RETURN_IF_NULL(context, StubReturn());
+    UNIMPLEMENTED();
+    return StubReturn();
 }
 
 /**
- @Status Caveat
- @Notes Manually converts CMYK to RGB, and does not involve the colorspace.
+ @Status Interoperable
 */
-void CGContextSetCMYKFillColor(CGContextRef context, CGFloat cyan, CGFloat magenta, CGFloat yellow, CGFloat black, CGFloat alpha) {
+void CGContextAddLines(CGContextRef context, const CGPoint* points, unsigned count) {
     NOISY_RETURN_IF_NULL(context);
-    CGContextSetRGBFillColor(context,
-                             (1.0f - cyan) * (1.0f - black),
-                             (1.0f - magenta) * (1.0f - black),
-                             (1.0f - yellow) * (1.0f - black),
-                             alpha);
+
+    if (!count || !points) {
+        return;
+    }
+
+    // TODO(DH) GH#1066 Switch to CGPathAddLines(context->path, points, count);
+    CGContextMoveToPoint(context, points[0].x, points[0].y);
+    for (unsigned int i = 1; i < count; i++) {
+        CGContextAddLineToPoint(context, points[i].x, points[i].y);
+    }
 }
 #pragma endregion
 
-#pragma region Graphics States
+#pragma region Global State - Path Queries
 /**
  @Status Interoperable
+ @Notes
 */
-void CGContextSaveGState(CGContextRef context) {
-    NOISY_RETURN_IF_NULL(context);
-    auto& oldState = context->CurrentGState();
-
-    // This uses the drawing state's copy constructor.
-    context->GStateStack().emplace(oldState);
+CGPathRef CGContextCopyPath(CGContextRef context) {
+    NOISY_RETURN_IF_NULL(context, StubReturn());
+    UNIMPLEMENTED();
+    return StubReturn();
+}
 
-    auto factory = context->Factory();
-    ComPtr<ID2D1DrawingStateBlock> drawingState;
-    FAIL_FAST_IF_FAILED(factory->CreateDrawingStateBlock(&drawingState));
+/**
+ @Status Stub
+ @Notes
+*/
+CGPoint CGContextGetPathCurrentPoint(CGContextRef context) {
+    NOISY_RETURN_IF_NULL(context, StubReturn());
+    UNIMPLEMENTED();
+    return StubReturn();
+}
 
-    context->RenderTarget()->SaveDrawingState(drawingState.Get());
-    oldState.d2dState = drawingState;
+/**
+ @Status Stub
+ @Notes
+*/
+bool CGContextPathContainsPoint(CGContextRef context, CGPoint point, CGPathDrawingMode mode) {
+    NOISY_RETURN_IF_NULL(context, StubReturn());
+    UNIMPLEMENTED();
+    return StubReturn();
 }
+#pragma endregion
```

Comment:
@DHowett-MSFT other than arranging the file to be more logical, are there real code changes that need to be reviewed?